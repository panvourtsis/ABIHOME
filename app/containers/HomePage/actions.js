import {
  LOAD_IMAGES,
  LOAD_IMAGES_SUCCESS,
  LOAD_IMAGES_ERROR,
  SELECT_IMAGE,
} from './constants';


/**
 * Changes the state of loading images
 * @param  {boolean} imagesLoaded - The boolean type of the state
 * @return {object} An action object with a type of LOAD_IMAGES
 */
export function loadImages(areImagesLoaded) {
  return {
    type: LOAD_IMAGES,
    imagesLoaded: areImagesLoaded
  };
}

/**
 * Dispatched when loading the images succesfully
 * @param  {array} images - The array of images fetched
 * @return {object} An action object with a type of LOAD_IMAGES_SUCCESS
 */
export function imagesLoaded(images) {
  return {
    type: LOAD_IMAGES_SUCCESS,
    images,
  };
}

/**
 * Dispatched when loading the images fails
 * @return {object} An action object with a type of LOAD_IMAGES_ERROR
 */
export function imagesLoadedError() {
  return {
    type: LOAD_IMAGES_ERROR,
  };
}

/**
 * Dispatched when selecting an image from the list
 * @param  {string} image - The image clicked
 * @return {object} An action object with a type of SELECT_IMAGE
 */
export function selectImage(image) {
  return {
    type: SELECT_IMAGE,
    image,
  };
}
