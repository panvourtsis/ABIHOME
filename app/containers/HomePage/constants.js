export const CHANGE_USERNAME = 'abihome/Home/CHANGE_USERNAME';
export const LOAD_IMAGES = 'abihome/Home/LOAD_IMAGES';
export const LOAD_IMAGES_SUCCESS = 'abihome/Home/LOAD_IMAGES_SUCCESS';
export const LOAD_IMAGES_ERROR = 'abihome/Home/LOAD_IMAGES_ERROR';
export const SELECT_IMAGE = 'abihome/Home/SELECT_IMAGE';
