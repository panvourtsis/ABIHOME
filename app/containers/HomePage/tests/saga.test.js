import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_IMAGES } from '../constants';
import { imagesLoaded, imagesLoadedError } from '../actions';

import homeData, { getImages } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('getImages Saga', () => {
  let getImagesGenerator;

  beforeEach(() => {
    getImagesGenerator = getImages();

    const selectDescriptor = getImagesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the imagesLoaded action if it requests the data successfully', () => {
    const response = [
      'first image',
      'second image',
      'third image'
    ];
    const putDescriptor = getImagesGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(imagesLoaded(response)));
  });

  it('should call the imagesLoadedError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getImagesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(imagesLoadedError()));
  });
});

describe('homeData Saga', () => {
  const homeDataSaga = homeData();

  it('should start task to watch for LOAD_IMAGES action', () => {
    const takeLatestDescriptor = homeDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_IMAGES, getImages));
  });
});
