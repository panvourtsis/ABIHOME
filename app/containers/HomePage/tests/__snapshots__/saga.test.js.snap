// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getImages Saga should call the imagesLoadedError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://weyveed.herokuapp.com/test/images",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getImages Saga should dispatch the imagesLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://weyveed.herokuapp.com/test/images",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
